common/djangoapps/student/forms.py

```diff
@@ -47,7 +49,20 @@ def clean_email(self):
         """
         email = self.cleaned_data["email"]
         #The line below contains the only change, removing is_active=True
-        self.users_cache = User.objects.filter(email__iexact=email)
+        if settings.FEATURES.get('APPSEMBLER_MULTI_TENANT_EMAILS', False):
+            current_site = get_current_site()
+            if current_site.id == settings.SITE_ID:
+                raise NotImplementedError('email_exists_or_retired expecting multi-tenant site')
+            # Using `get` is expected to fail when multiple-orgs found for a site
+            # TODO: Handle both MultipleObjectsReturned and DoesNotExist in a better way
+            current_org = current_site.organizations.get()
+            try:
+                self.users_cache = [current_org.userorganizationmapping_set.get(user__email__iexact=email).user]
+            except UserOrganizationMapping.DoesNotExist:
+                self.users_cache = []
```

Comment:
I think this would save us the exception and match the exact functionality by edX which is to find multiple emails.

```suggestion
            found_mappings = UserOrganizationMapping.objects.filter(
                user__email__iexact=email,
                organization=current_org,
                is_active=True,
            )
            self.users_cache = [mapping.user for mapping in found_mappings]
```